android/src/main/java/com/mattermost/networkclient/GenericClientModule.kt

```diff
@@ -0,0 +1,93 @@
+package com.mattermost.networkclient
+
+import com.facebook.react.bridge.*
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+
+
+class GenericClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+
+    var client = OkHttpClient();
+
+    override fun getName(): String {
+        return "GenericClient"
+    }
+
+    private fun parseResponse(response: Response): WritableMap {
+        val headers = Arguments.createMap();
+        response.headers.forEach{ k -> headers.putString(k.first, k.second) }
+
+        val map = Arguments.createMap()
+        map.putMap("headers", headers)
+        map.putString("data", response.body!!.string())
+        map.putInt("code", response.code)
+        return map;
+    }
+
+    @ReactMethod
+    fun get(baseUrl: String, options: ReadableMap, promise: Promise){
+        val request = Request.Builder().url(baseUrl).build();
+        client.newCall(request).execute().use{ response ->
+            if (response.isSuccessful) {
+                val res = parseResponse(response);
+                return promise.resolve(res)
+            }
+            promise.reject("Unexpected code $response")
```

Comment:
I think you  should reject the promise but include the response even if its not successful so it bubbles up all the way to JS. Same thing with all the other methods